### ISSUE TITLE
m_spanningtree symbol errors on openbsd

### ISSUE THREAD

**Issue by benharri** [inspircd/inspircd#1635]
**Description**

m_spanningtree crashes insp3 on openbsd.

**Steps to reproduce the issue:**

1. `git checkout v3.0.0`
2. `doas pkg_add gnutls gmake`
3. `gmake install`
4. `./inspircd start`

**Describe the results you received:**

`inspircd:/home/ben/inspircd/run/modules/m_spanningtree.so: undefined symbol '_ZN11CommandSave14SavedTimestampE'`

**Describe the results you expected:**

inspircd starts normally

**Comment by gbmor** [comment]
I am also experiencing this issue.

I have tried compiling with the following:

g++ 4.2.1 20070719
ld.bfd 2.17 (GNU ld)
clang++ 6.0.0
ld.lld 6.0.0

And in all combinations, InspIRCd dies when loading m_spanningtree.so, with the same error message as above.

**Comment by SaberUK** [comment]
@gbmor GCC 4.2.1 is too old to build InspIRCd v3 with. Please build with a more recent compiler.

@benharri Please can you confirm the compiler and compiler version you are building with.

**Comment by gbmor** [comment]
@SaberUK In my post, I mentioned it was also tried with clang++ 6.0.0

**Comment by benharri** [comment]
I spent a couple hours building gcc8 from ports this evening and it seems to be working now that I built using modern gcc. 

linking m_spanningtree still fails when building with clang.

**Comment by SaberUK** [comment]
Please can you post the contents of CPPFLAGS, CXXFLAGS, LDFLAGS in the environment when buuilding with clang.

OpenBSD have historically done terrible things to their version of Clang (see also: #410) in the name of """securit"y"" which have broken building/running InspIRCd entirely.

**Comment by gbmor** [comment]
```
ahriman@tilde ~ $ clang++ --version
OpenBSD clang version 6.0.0 (tags/RELEASE_600/final) (based on LLVM 6.0.0)
Target: amd64-unknown-openbsd6.4
Thread model: posix
InstalledDir: /usr/bin
```
`CPPFLAGS`, `CXXFLAGS`, and `LDFLAGS` have not been set.

I compiled g++6 from ports, however, the same issue presented itself. I copied the gcc8 meta information from the ports tree intended for OpenBSD `6.5` on a whim and I was able to get g++ 8.3.0 compiled on my `6.4` machine, then successfully built InspIRCd v3.0.0 using it. A runtime issue has presented itself, but it's not related to this issue.

It looks like a successful workaround will be to compile with gcc8, from the OpenBSD `6.5` ports tree, specifically.

**Comment by SaberUK** [comment]
A speculative fix for this issue, find the following line in `src/modules/m_spanningtree/commands.h`:

```cpp
class CommandSave : public ServerCommand
```

And replace it with this:

```cpp
class CoreExport CommandSave : public ServerCommand
```

**Comment by SaberUK** [comment]
@benharri Please can you confirm whether this fixes the issue.

**Comment by SaberUK** [comment]
Having diagnosed this with @benharri I think this is an issue with OpenBSDs compilers being completely broken. We can't really work around this, sorry. :(

**Comment by benharri** [comment]
The speculative fix is working when built with gcc 8.3.0.
```
class CoreExport CommandSave : public ServerCommand
```

There was an additional error we found that was related to a malformed response from the DNS resolver I was using.
